### PR TITLE
Team refresh

### PR DESCRIPTION
**Pull request by buoyad**
- Use `GetAnnotatedTeam`
- Split `teamMeta` and `teamDetails` so it is more straightforward to know what we've loaded
- Impl team reload strategy. Keep list of "interested teams" and eagerly reload when a version increases and we have loaded the team before.
- Various fixes to `GetAnnotatedTeam`
- Remove old separate flows for loading publicity settings

TODO
- [x] Verify refresh strategy works for places we were calling `getDetails` before.
- [x] Changing team settings borked
- [x] Change uses of old publicity settings to details

cc @heronhaye

### CODE REVIEW

**Comment by heronhaye** [comment]
Go code LGTM, thanks for fixing

**Comment by buoyad** [comment]
Ready for review! cc @keybase/y2ksquad

File: shared/actions/teams.tsx [hunk 1]
```diff
@@ -500,87 +496,18 @@ async function createNewTeamFromConversation(
...
-  } catch (e) {
-    logger.error(e)
-  } finally {
-    const loadingKey = action.payload.clearInviteLoadingKey
-    if (loadingKey) {
-      yield Saga.put(TeamsGen.createSetTeamLoadingInvites({isLoading: false, loadingKey, teamname}))
-    }
+// We reload teams eagerly on updates if any of their versions increases and
+// we've previously loaded the team in this session.
+const interestedTeams: Set<Types.TeamID> = new Set()
```
Comment (jakob223) [inline shared/actions/teams.tsx]: why keep this as a local variable instead of in the redux state?

File: shared/actions/teams.tsx [hunk 2]
```diff
@@ -941,14 +836,14 @@ const setMemberPublicity = async (_: TypedState, action: TeamsGen.SetMemberPubli
       Constants.teamWaitingKey(teamname)
     )
     return [
-      TeamsGen.createGetDetails({teamname}),
+      // TeamsGen.createGetDetails({teamname}),
       // The profile showcasing page gets this data from teamList rather than teamGet, so trigger one of those too.
```
Comment (jakob223) [inline shared/actions/teams.tsx]: can we delete this commented out code (and below)?

File: shared/actions/teams.tsx [hunk 3]
```diff
@@ -1061,7 +955,7 @@ function* setPublicity(state: TypedState, action: TeamsGen.SetPublicityPayload,
 
   const results = yield Saga.all(calls)
   // TODO delete this getDetails call when CORE-7125 is finished
```
Comment (jakob223) [inline shared/actions/teams.tsx]: 🗑

File: shared/profile/showcase-team-offer/index.tsx [hunk 4]
```diff
@@ -98,17 +98,17 @@ const ShowcaseTeamOffer = (props: Props) => {
...
-            canShowcase={teamDetails.allowPromote}
-            isExplicitMember={teamDetails.isMember}
-            name={teamDetails.teamname}
-            isOpen={teamDetails.isOpen}
-            membercount={teamDetails.memberCount}
-            onPromote={promoted => props.onPromote(teamDetails.teamname, promoted)}
-            showcased={teamDetails.showcasing}
-            waiting={!!props.waiting[teamWaitingKey(teamDetails.teamname)]}
+            key={teamMeta.id}
+            canShowcase={teamMeta.allowPromote || ['admin', 'owner'].includes(teamMeta.role)}
```
Comment (jakob223) [inline shared/profile/showcase-team-offer/index.tsx]: was this a bug before?

File: shared/constants/teams.tsx [hunk 5]
```diff
@@ -726,3 +764,13 @@ export const getCanPerform = (state: TypedState, teamname: Types.Teamname): Type
 
 export const getCanPerformByID = (state: TypedState, teamID: Types.TeamID): Types.TeamOperations =>
   deriveCanPerform(state.teams.teamRoleMap.roles.get(teamID))
+
+// Don't allow version to roll back
```
Comment (jakob223) [inline shared/constants/teams.tsx]: 👍

File: shared/teams/team/container.tsx [hunk 6]
```diff
@@ -28,23 +28,28 @@ const mapStateToProps = (state: Container.TypedState, ownProps: OwnProps) => {
   const selectedTab = ownProps.selectedTab || 'members'
 
   return {
-    _teamnameTodoRemove: Constants.getTeamDetails(state, teamID).teamname,
+    _teamnameTodoRemove: Constants.getTeamMeta(state, teamID).teamname,
```
Comment (jakob223) [inline shared/teams/team/container.tsx]: nice name lol

File: shared/profile/showcase-team-offer/index.tsx (hunk 4 above)
Comment (buoyad) [reply shared/profile/showcase-team-offer/index.tsx]: yes

**Pull request closed without merging**